### DISCUSSION TITLE
Add web thermostat interface to the web interface

### DISCUSSION THREAD

**Discussion by matteopedani** [emsesp/EMS-ESP32#1065] [Ideas]
As my Junkers boiler  when sense a EMS device stay alwais ON, looking  for a bootstap sequence. The result if you don' t send the sequence   you  can't write commands but only read command. 

Will be great add a small web page on  the interface to program EMS-ESP32  act like   thermostat. 

In the web interface you will chose:
  
1) the boiler to pilot 
2) the boot sequence
3) switch ON/OFF  the boiler 
4) read if the boiler is ON/OFF 
5) Set the  temperatures 
6) read the temperature you can set
7) add a simple timer  ON/OFF  scheduler  
 
 Just simple interface to test a boostrap the boiler  without any other  device or computer program

**Comment by proddy** [comment]
I'd like to learn more about this sequence. Could you explain more what you're seeing?

on the Web thermostat control, this is indeed possible but is not aligned with the goal of EMS-ESP which was to build an interface between the EMS protocol and the iot world. We have always refrained from trying to mimic controllers (like a thermostat) and suggest using automation tools like Domoticz, Home Assistant, OpenHAB, etc for this. There are examples of how to do this in the documentation.

**Reply by matteopedani** [reply]
Can you attach a link to an example, because I did't find it. 
But my idea was just a easy way to store and send commands on the right sequence  to bootstrap and  pilot a boiler, not just to act as thermostat. 
 Thank's

**Reply by proddy** [reply]
ok, you did say "Will be great add a small web page on the interface to program EMS-ESP32 act like thermostat". Look in the doc for EMS-ESP and you'll find some examples. For the boot strap it's just easier to create a python script, or shell, or a job in Home Assistant to automate this. The EMS-ESP code is almost full, with Flash on a 4MB chip now at 99% which means we can't add any more features until we have the nextgen version out.

**Comment by matteopedani** [comment]
OK, thank's

I think will be good add some space on documentation, to how to  bootstrap
boilers.
Can I change the documentation?

Il giorno mar 21 feb 2023 alle ore 13:52 Proddy ***@***.***>
ha scritto:

> ok, you did say "Will be great add a small web page on the interface to
> program EMS-ESP32 act like thermostat". Look in the doc for EMS-ESP and
> you'll find some examples. For the boot strap it's just easier to create a
> python script, or shell, or a job in Home Assistant to automate this. The
> EMS-ESP code is almost full, with Flash on a 4MB chip now at 99% which
> means we can't add any more features until we have the nextgen version out.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/emsesp/EMS-ESP32/discussions/1065#discussioncomment-5063831>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEUI6BJBO5OBZKG2BRE5MVTWYS3BVANCNFSM6AAAAAAVA6JMQA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
*Matteo Pedani*

phone +39 069415152

**Reply by proddy** [reply]
yes, fork the repo https://github.com/emsesp/docs, change `/docs/tips-and-tricks.md` and push a PR

If you want an example of how to control EMS-ESP from a script look at  https://github.com/emsesp/EMS-ESP32/blob/tech-upgrade/scripts/run_memory_test.py
